pkg/eestream/rs_test.go

```diff
@@ -157,18 +158,18 @@ func TestRSLateEOF(t *testing.T) {
 		{4 * 1024, 1024, 1, 1, 0, false},
 		{4 * 1024, 1024, 1, 1, 1, true},
 		{4 * 1024, 1024, 1, 2, 0, false},
-		{4 * 1024, 1024, 1, 2, 1, false},
+		// {4 * 1024, 1024, 1, 2, 1, false},
 		{4 * 1024, 1024, 1, 2, 2, true},
 		{4 * 1024, 1024, 2, 4, 0, false},
 		{4 * 1024, 1024, 2, 4, 1, false},
-		{4 * 1024, 1024, 2, 4, 2, false},
+		// {4 * 1024, 1024, 2, 4, 2, false},
 		{4 * 1024, 1024, 2, 4, 3, true},
 		{4 * 1024, 1024, 2, 4, 4, true},
 		{6 * 1024, 1024, 3, 7, 0, false},
 		{6 * 1024, 1024, 3, 7, 1, false},
 		{6 * 1024, 1024, 3, 7, 2, false},
-		{6 * 1024, 1024, 3, 7, 3, false},
-		{6 * 1024, 1024, 3, 7, 4, false},
+		// {6 * 1024, 1024, 3, 7, 3, false},
+		// {6 * 1024, 1024, 3, 7, 4, false},
```

Comment:
what happened with these tests?

Reply:
The outcome of these tests are now unreliable due to the random behavior of the selector. For example, let's look at the test case with 3 required, 7 total, and 3 problematic readers returning just EOF. There are two possible scenarios due to the selector:
1) The problematic readers will be selected first, returning EOF. 3 EOF are 3 required readers returning EOF. Thus the decoder's Read() will return EOF.
2) The good readers will be selected first, returning good data. 4 good readers will decode the data successfully.
As it cannot be determined which of the two scenarios will happen, the assertion of the decoded data will fail.

Reply:
okay, well it'd be good to either make the tests more robust or get rid of them entirely. i also think this is another good reason for why we should only return EOF when we hit the expected size for the DecodeReaderSized

Reply:
Yes, I am looking at the expected size thing.